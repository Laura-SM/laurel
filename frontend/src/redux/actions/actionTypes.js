export default {
  LOAD_PLANTS: 'LOAD_PLANTS',
  ADD_PLANT: 'ADD_PLANT',
  UPDATE_PLANT: 'UPDATE_PLANT',
  DELETE_PLANT: 'DELETE_PLANT',
  LOAD_PLANT: 'LOAD_PLANT',
  PLANTS_ERROR: 'PLANTS_ERROR',

  GET_USER: 'GET_USER',
  USER_ERROR: 'USER_ERROR',
};
