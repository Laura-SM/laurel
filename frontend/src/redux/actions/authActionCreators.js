import axios from 'axios';
import actionTypes from './actionTypes';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function signInUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/signin`, user);
      dispatch({
        type: actionTypes.SIGNIN_USER,
        userTokens: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}

export function signUpUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/signup`, user);
      dispatch({
        type: actionTypes.SIGNUP_USER,
        user: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}
