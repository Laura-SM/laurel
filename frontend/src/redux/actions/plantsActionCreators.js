import axios from 'axios';
import actionTypes from './actionTypes';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function loadPlants() {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/plants`);
      dispatch({
        type: actionTypes.LOAD_PLANTS,
        plants: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function addPlant(plant) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/plants`, plant);
      dispatch({
        type: actionTypes.ADD_PLANT,
        plant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function deletePlant(plantId) {
  return async dispatch => {
    try {
      await axios.delete(`${url}/plants/${plantId}`);
      dispatch({
        type: actionTypes.DELETE_PLANT,
        plantId,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function updatePlant(plant) {
  return async dispatch => {
    try {
      const {data} = await axios.put(`${url}/plants/${plant._id}`, plant);
      dispatch({
        type: actionTypes.UPDATE_PLANT,
        plant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function loadPlant(plantId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/plants/${plantId}`);
      dispatch({
        type: actionTypes.LOAD_PLANT,
        selectedPlant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}
