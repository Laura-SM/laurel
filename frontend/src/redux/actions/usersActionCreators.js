import axios from 'axios';
import actionTypes from './actionTypes';

const url = 'http://192.168.8.104:2021/login';

export function loadUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(url, user);
      dispatch({
        type: actionTypes.SIGNIN_USER,
        userTokens: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}
