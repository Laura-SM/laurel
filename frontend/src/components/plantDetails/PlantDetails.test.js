import 'react-native';
import React from 'react';
import PlantDetails from './index';
import * as actions from '../../redux/actions/plantsActionCreators';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a PlantDetails component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;
  let route = {
    params: {plantId: '1a'},
  };

  beforeEach(() => {
    myStore = mockStore({
      selectedPlant: {
        name: 'kentia',
        _id: '1a',
        image: 'img.com',
        room: 'bedroom',
      },
    });
    navigation = {
      goBack: jest.fn(),
    };
  });

  describe('When invoques PlantDetails function', () => {
    test('Should render a plant with detail', () => {
      const detail = render(
        <Provider store={myStore}>
          <PlantDetails route={route} />
        </Provider>,
      );
      expect(detail).toMatchSnapshot();
    });
  });

  describe('When goBack button is pressed', () => {
    test('Then navigation.goBack is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <PlantDetails route={route} navigation={navigation} />
        </Provider>,
      );
      const goBack = getByTestId('goBackButton');
      fireEvent.press(goBack);

      expect(navigation.goBack).toHaveBeenCalled();
    });
  });
});
