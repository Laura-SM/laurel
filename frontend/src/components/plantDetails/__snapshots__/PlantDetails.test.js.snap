// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PlantDetails component When invoques PlantDetails function Should render a plant with detail 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "margin": 20,
        }
      }
    >
      kentia
    </Text>
  </View>
  <Text
    style={
      Object {
        "borderBottomColor": "#93B1A7",
        "borderBottomWidth": 2,
        "fontSize": 18,
        "fontStyle": "italic",
        "marginHorizontal": 20,
        "paddingBottom": 10,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "img.com",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 250,
          "margin": 20,
          "width": 170,
        }
      }
    />
    <View
      style={
        Object {
          "height": 250,
          "justifyContent": "space-around",
          "width": 170,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/icons/dangerous24.png",
            }
          }
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginHorizontal": 20,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#B8E0D2",
          "borderRadius": 50,
          "height": 60,
          "justifyContent": "center",
          "margin": 20,
          "opacity": 1,
          "width": 60,
        }
      }
      testID="goBackButton"
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/icons/goBack24.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
